### PR TITLE
TGUI Preferences Menu + total rewrite of the preferences backend

### PR DESCRIPTION
**Pull request by ThatLing**
# Document the changes in your pull request
So this, is a big one and will break a lot of stuff.

Mainly ports https://github.com/tgstation/tgstation/pull/61313, which a whole bunch of extra stuff because we are so outdated. Currently still heavily WIP, so no screenshots yet. If you want to know what it looks like, go to the original PR.

Full list of ported PRs, credits to the original authors:
- https://github.com/tgstation/tgstation/pull/58483
- https://github.com/tgstation/tgstation/pull/58593 (partly)
- https://github.com/tgstation/tgstation/pull/59914
- https://github.com/tgstation/tgstation/pull/61313
- https://github.com/tgstation/tgstation/pull/61542
- https://github.com/tgstation/tgstation/pull/61586
- https://github.com/tgstation/tgstation/pull/61764
- https://github.com/tgstation/tgstation/pull/61942
- https://github.com/tgstation/tgstation/pull/61957
- https://github.com/tgstation/tgstation/pull/61980 (TODO)
- https://github.com/tgstation/tgstation/pull/62845
- https://github.com/tgstation/tgstation/pull/63200
- https://github.com/tgstation/tgstation/pull/63370
- https://github.com/tgstation/tgstation/pull/63503
- https://github.com/tgstation/tgstation/pull/63554 (partly)
- https://github.com/tgstation/tgstation/pull/64610
- https://github.com/tgstation/tgstation/pull/65140
- https://github.com/tgstation/tgstation/pull/65645
- https://github.com/tgstation/tgstation/pull/66317
- https://github.com/tgstation/tgstation/pull/69454
- https://github.com/tgstation/tgstation/pull/69635
- https://github.com/tgstation/tgstation/pull/69852
- https://github.com/tgstation/tgstation/pull/70577

# Changelog

:cl: Mothblocks & a lot of people from /tg/
rscadd: The preferences menu has been completely rewritten in tgui.
tweak: The "Preferences" tab has been removed, the menu verbs have been moved to the OOC tab
tweak: The "Stop Sounds" verb has been moved to OOC.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/species_types/ethereal.dm [hunk 1]
```diff
@@ -183,3 +186,55 @@
...
+			SPECIES_PERK_ICON = "lightbulb",
+			SPECIES_PERK_NAME = "Disco Ball",
+			SPECIES_PERK_DESC = "Ethereals passively generate their own light.",
+		),
+		list(
+			SPECIES_PERK_TYPE = SPECIES_NEUTRAL_PERK,
+			SPECIES_PERK_ICON = "gem",
+			SPECIES_PERK_NAME = "Crystal Core",
+			SPECIES_PERK_DESC = "The Ethereal's heart will encase them in crystal should they die, returning them to life after a time - \
+				at the cost of a permanent brain trauma.",
```
Comment (Moltijoe) [inline code/modules/mob/living/carbon/human/species_types/ethereal.dm:230]: they don't do this here

File: code/modules/mob/living/carbon/human/species_types/felinid.dm [hunk 2]
```diff
@@ -192,3 +192,60 @@
...
+			SPECIES_PERK_TYPE = SPECIES_NEGATIVE_PERK,
+			SPECIES_PERK_ICON = "assistive-listening-systems",
+			SPECIES_PERK_NAME = "Sensitive Hearing",
+			SPECIES_PERK_DESC = "Felinids are more sensitive to loud sounds, such as flashbangs.",
+		),
+		list(
+			SPECIES_PERK_TYPE = SPECIES_NEGATIVE_PERK,
+			SPECIES_PERK_ICON = "shower",
+			SPECIES_PERK_NAME = "Hydrophobia",
+			SPECIES_PERK_DESC = "Felinids don't like getting soaked with water.",
```
Comment (Moltijoe) [inline code/modules/mob/living/carbon/human/species_types/felinid.dm:247]: i don't think they have any of these perks here, i will need to double check

File: code/modules/mob/living/carbon/human/species_types/flypeople.dm [hunk 3]
```diff
@@ -31,3 +32,57 @@
...
+			SPECIES_PERK_TYPE = SPECIES_NEGATIVE_PERK,
+			SPECIES_PERK_ICON = "fist-raised",
+			SPECIES_PERK_NAME = "Insectoid Biology",
+			SPECIES_PERK_DESC = "Fly swatters will deal significantly higher amounts of damage to a Flyperson.",
+		),
+		list(
+			SPECIES_PERK_TYPE = SPECIES_NEGATIVE_PERK,
+			SPECIES_PERK_ICON = "sun",
+			SPECIES_PERK_NAME = "Radial Eyesight",
+			SPECIES_PERK_DESC = "Flypeople can be flashed from all angles.",
```
Comment (Moltijoe) [inline code/modules/mob/living/carbon/human/species_types/flypeople.dm:77]: we don't have sight angles so anyone can be flashed from all angles

File: code/modules/mob/living/carbon/human/species_types/flypeople.dm [hunk 4]
```diff
@@ -31,3 +32,57 @@
...
+		),
+		list(
+			SPECIES_PERK_TYPE = SPECIES_NEGATIVE_PERK,
+			SPECIES_PERK_ICON = "briefcase-medical",
+			SPECIES_PERK_NAME = "Weird Organs",
+			SPECIES_PERK_DESC = "Flypeople take specialized medical knowledge to be \
+				treated. Their organs are disfigured and organ manipulation can be interesting...",
```
Comment (Moltijoe) [inline code/modules/mob/living/carbon/human/species_types/flypeople.dm:84]: i don't _THINK_ they have this here

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 5]
```diff
@@ -1548,3 +1563,46 @@
...
+/datum/species/golem/cloth/create_pref_unique_perks()
+	var/list/to_add = ..()
+
+	to_add += list(list(
+		SPECIES_PERK_TYPE = SPECIES_POSITIVE_PERK,
+		SPECIES_PERK_ICON = "recycle",
+		SPECIES_PERK_NAME = "Reformation",
+		SPECIES_PERK_DESC = "A boon quite similar to Ethereals, Mummies collapse into \
+			a pile of bandages after they die. If left alone, they will reform back \
+			into themselves. The bandages themselves are very vulnerable to fire.",
```
Comment (Moltijoe) [inline code/modules/mob/living/carbon/human/species_types/golems.dm:1590]: not "similar to ethereals"

File: code/modules/mob/living/carbon/human/species_types/humans.dm [hunk 6]
```diff
@@ -29,3 +29,54 @@
...
+			SPECIES_PERK_DESC = "The AI and their cyborgs are, by default, subservient only \
+				to humans. As a human, silicons are required to both protect and obey you.",
+		))
+
+	if(CONFIG_GET(flag/enforce_human_authority))
+		to_add += list(list(
+			SPECIES_PERK_TYPE = SPECIES_POSITIVE_PERK,
+			SPECIES_PERK_ICON = "bullhorn",
+			SPECIES_PERK_NAME = "Chain of Command",
+			SPECIES_PERK_DESC = "Nanotrasen only recognizes humans for command roles, such as Captain.",
```
Comment (Moltijoe) [inline code/modules/mob/living/carbon/human/species_types/humans.dm:79]: only for HoP and Captain, other command roles have some non-human species.
Maybe include better pay as one of the perks

File: code/modules/mob/living/carbon/human/species_types/jellypeople.dm [hunk 7]
```diff
@@ -67,6 +68,21 @@
...
+// They may not be roundstart but in the unlikely event they become one might as well not leave a glaring issue open.
+/datum/species/jelly/create_pref_blood_perks()
+	var/list/to_add = list()
+
+	to_add += list(list(
+		SPECIES_PERK_TYPE = SPECIES_NEUTRAL_PERK,
+		SPECIES_PERK_ICON = "tint",
+		SPECIES_PERK_NAME = "Jelly Blood",
+		SPECIES_PERK_DESC = "[plural_form] don't have blood, but instead have toxic [initial(exotic_blood.name)]! \
+			Jelly is extremely important, as losing it will cause you to lose limbs. Having low jelly will make medical treatment very difficult.",
```
Comment (Moltijoe) [inline code/modules/mob/living/carbon/human/species_types/jellypeople.dm:81]: should also mention inverted toxin damage

File: yogstation/code/modules/mob/living/carbon/human/species_types/preternis/preternis.dm [hunk 8]
```diff
@@ -7,6 +7,7 @@ adjust_charge - take a positive or negative value to adjust the charge level
 
 /datum/species/preternis
 	name = "Preternis"
+	plural_form = "Preternis"
```
Comment (Moltijoe) [inline yogstation/code/modules/mob/living/carbon/human/species_types/preternis/preternis.dm:10]: Preterni

File: yogstation/code/modules/mob/living/carbon/human/species_types/preternis/preternis.dm [hunk 9]
```diff
@@ -273,3 +274,25 @@ adjust_charge - take a positive or negative value to adjust the charge level
...
+
+/datum/species/preternis/get_species_lore()
+	return list(
+		"TODO: This is preternis lore"
+	)
+
+/datum/species/preternis/create_pref_unique_perks()
+	var/list/to_add = list()
+
+	// TODO
```
Comment (Moltijoe) [inline yogstation/code/modules/mob/living/carbon/human/species_types/preternis/preternis.dm:289]: **probably something about** 
self surgery and self repair from being metal
Emp immunity
Eyes functioning to give night vision
Crippling weakness to water
Can't wear shoes, but have innate magboot abilities
purges all chems after a short time
reduced oxygen needed to breathe

Ideally try to roll some of these together to reduce bloat

**Comment by Cartlord** [comment]
I don't like the idea of removing the Preferences tab, because that's packed already - moving it to the OOC tab will make that much worse

**Comment by JamieD1** [comment]
MOVING TO YOGS BRANCH

**Pull request merged**